lib/msal-browser/src/app/ClientApplication.ts

```diff
@@ -166,7 +166,12 @@ export abstract class ClientApplication {
                 return null;
             } else {
                 BrowserUtils.clearHash();
-                return responseHash;
+                if (this.interactionInProgress()) {
```

Comment:
We should do this check at the beginning, so that we don't have to do any cache lookups.

Reply:
Issue is that we want to clear the hash if we aren't expecting it but we don't want to clear it if interactionType != Redirect. So I think it needs to stay here unless anyone has a suggestion for solving that problem.

Reply:
> Issue is that we want to clear the hash if we aren't expecting it

I don't think we should be clearing the hash if we aren't expecting it, that seems like it could break non-msal hashes.

Reply:
That's why this check is being done here. Because by this point we've already determined that the hash **is** an msal hash

Reply:
Resolved offline. Interaction check only checks for this `clientId`, in theory a parallel PCA using a different clientId could have initiated the request in which case we would not want to clear the hash. Moved to the top.

Reply:
Resolved offline - multiple app use cases may break if we clear, so probably better to move the check before we clear.